server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java

```diff
@@ -301,6 +484,36 @@ ComponentRegistry getCacheRegistry(String cacheName) {
       return server.getCacheRegistry(cacheName);
    }
 
+   /**
+    * Validates if the value read is still valid and the write operation can proceed.
+    */
+   private boolean isValid(TransactionWrite write, AdvancedCache<byte[], byte[]> readCache) {
+      if (write.skipRead()) {
+         if (isTrace) {
+            log.tracef("Operation %s wasn't read.", write);
+         }
+         return true;
+      }
+      CacheEntry<byte[], byte[]> entry = readCache.getCacheEntry(write.key);
+      if (write.wasNonExisting()) {
+         if (isTrace) {
+            log.tracef("Key didn't exist for operation %s. Entry is %s", write, entry);
+         }
+         return entry == null;
+      }
+      if (isTrace) {
+         log.tracef("Checking version for operation %s. Entry is %s", write, entry);
+      }
+      return entry != null && write.versionRead == extractVersion(entry.getMetadata().version());
```

Comment:
I'll use the extractVersion until the issue is fixed. Otherwise, it won't be compliant since the getWithMetadata uses this method to send the version back to the client (and it is this version that is sent to validate)